scripts/check.sh

```diff
@@ -34,6 +34,11 @@ check-protos-compatible() {
     # If the output message is encountered, please add proto.lock to git as well.
     git diff scripts/proto.lock | cat
     git diff --quiet scripts/proto.lock
+    if [ $? -ne 0 ]; then
```

Comment:
As described in this pr https://github.com/pingcap/kvproto/pull/1058
I modified it to force the update and only make it like a hint, I think it's better to keep it like a hint. Rather than return error to confuse developers and it will also increase developers' learning costs.